phpunit/class-block-library-comment-template-test.php

```diff
@@ -120,6 +120,61 @@ function test_build_comment_query_vars_from_block_no_context() {
 		);
 	}
 
+	/**
+	 * Test that if pagination is set to display the last page by default (i.e. newest comments),
+	 * the query is set to look for page 1 (rather than page 0, which would cause an error).
+	 *
+	 * Regression: https://github.com/WordPress/gutenberg/issues/40758.
+	 *
+	 * @covers ::build_comment_query_vars_from_block
+	 */
+	function test_build_comment_query_vars_from_block_pagination_with_no_comments() {
+		$comments_per_page     = get_option( 'comments_per_page' );
+		$default_comments_page = get_option( 'default_comments_page' );
+
+		update_option( 'comments_per_page', 50 );
+		update_option( 'previous_default_page', 'newest' );
+
+		$post_without_comments = self::factory()->post->create_and_get(
+			array(
+				'post_type'    => 'post',
+				'post_status'  => 'publish',
+				'post_name'    => 'fluffycat',
+				'post_title'   => 'Fluffy Cat',
+				'post_content' => 'Fluffy Cat content',
+				'post_excerpt' => 'Fluffy Cat',
+			)
+		);
+
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:comment-template --><!-- wp:comment-author-name /--><!-- wp:comment-content /--><!-- /wp:comment-template -->'
+		);
+
+		$block = new WP_Block(
+			$parsed_blocks[0],
+			array(
+				'postId' => $post_without_comments->ID,
+			)
+		);
+
+		$this->assertEquals(
+			array(
+				'orderby'       => 'comment_date_gmt',
+				'order'         => 'ASC',
+				'status'        => 'approve',
+				'no_found_rows' => false,
+				'post_id'       => $post_without_comments->ID,
+				'hierarchical'  => 'threaded',
+				'number'        => 50,
+				'paged'         => 1,
```

Comment:
Resyncing for the changes made when this code was committed into Core via changeset https://core.trac.wordpress.org/changeset/53336:

```suggestion
```

`'paged'` will not be set when there are no comments.